conans/model/conf.py

```diff
@@ -0,0 +1,152 @@
+import fnmatch
+from collections import OrderedDict
+
+from conans.errors import ConanException
+
+
+class _ConfModule(object):
```

Comment:
it looks like an in-house reimplementation of the standard [ConfigParser](https://docs.python.org/3/library/configparser.html), can't we just re-use it?

Reply:
Actually, it is implemented this way because the ConfigParser cannot work with the proposed syntax in Python2 (the ``delimiters`` argument that is necessary only works in Python3. Yes, we could fully drop support for Python2, but this needs to be in Conan 1.X, I'd prefer if it worked. 

Plus, the provided implementation provides more things:
- a much nicer access to the configuration, like ``self.conf["tools.microsoft.MSBuild"].verbosity`` that will not raise either for the module nor for the property. As this will be used in recipes and in helpers, I think it is important to have a friendly syntax.
- The necessary filter by modules, composition (updates), rebase (composition in inverse order)... The necessary parsing is actually only a very few lines, most of the code is not related at all to the "ini" parsing, as the format is completely straightforward: no sections at all. This is necessary to integrate with the profile ``[conf]`` section. I like very much this flat format, it will be much easier for any automation, commands, etc.

Reply:
I personally don't like the idea to invent our own data formats. can't we re-use xml/yaml/toml/json/whatever instead? I think any data could be serialized into existing formats, no point in inventing our own.
but if it's not a new format, but just a part of the conan profile format - it should belong to the profile format parser.

Reply:
That is the point, it should fit in the ``profile`` definition, so it cannot be yaml/json or whatever. And I really want to use the same syntax for the ``profile``, the ``conan.conf`` (alternative conan_conf.txt) and eventually, the command line syntax (it was implemented there, it was completely straightforward with this syntax). 

Parsing it in its own class is not a change of paradigm, it is what the profile_loader is already doing: https://github.com/conan-io/conan/blob/c7b9703d106de1a3ec7797a7875fac51bc76d8f6/conans/client/profile_loader.py#L212 for the ``OptionsValues`` class. If the syntax has to be read from the profile, the conan_conf and the command line, its parsing does not belong to the ``profile``, but to a common ground place, in this case, the new ``Conf`` classes are the most reasonable place to put it.